spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/core/SubscribingOperation.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.integration.core;
+
+import java.util.function.Consumer;
+
+/**
+ * Operations for subscribing to a destination.
+ *
+ * @author Warren Zhu
+ */
+public interface SubscribingOperation<D> {
+
+    /**
+     * Register a message consumer to a given destination.
+     *
+     * @return {@code true} if the consumer was subscribed or {@code false} if it
+     * was already subscribed.
+     */
+    boolean subscribe(String destination, Consumer<Iterable<D>> consumer);
```

Comment:
Why use `boolean` return value to indicate subscribe operation status? It requires the user to check the subscribe result with if/else conditions, why not we throw wrapped exception here instead to indicate the operation fails?

Reply:
In normal case, users doesn't need to check return result if they doesn't care the idempotent caused by subscribing same consumer multiple times. Exception has no benefits over this.

Reply:
If the user checks the subscription result and it's false, what can the user do with this?

Reply:
Nothing or the user has programming error.